Improve log when connection to kodi media player cannot be created

cvwillegen opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change

This makes it clear in the log files that a connection cannot be made to the Kodi server. This may save users time when they try to find why playing media does not work (any more).

<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [X] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

cvwillegen committed:
Make clearer in log files that connection to media player cannot be created.

JeffResc commented:
LGMT 👍

Not being able to connect is more of an error than a debug, in my opinion.

OnFreund commented:
This is going to print an error every 10 seconds when the device is off/disconnected/unavailable. I think it would be better to only print once until connected again.

JeffResc commented:
> This is going to print an error every 10 seconds when the device is off/disconnected/unavailable. I think it would be better to only print once until connected again.

Ah, I didn't catch that. @cvwillegen if you are not familiar with it, I would highly suggest taking a look at the [Integration Quality Scale](https://developers.home-assistant.io/docs/integration_quality_scale_index). It outlines different things you should take into consideration when creating or modifying an integration. Specifically on the Silver tier, it mentions "It will not print any exceptions nor will it fill the log with retry attempts." If possible, you should try to ensure as many of these specifications are satasfied.

cvwillegen committed:
Merge branch 'home-assistant:dev' into patch-1

cvwillegen committed:
Only give connection error once.

When connection is lost, only give an error message once. When connection is restored and lost again, give error message again.

cvwillegen commented:
> > This is going to print an error every 10 seconds when the device is off/disconnected/unavailable. I think it would be better to only print once until connected again.
> 
> Ah, I didn't catch that. @cvwillegen if you are not familiar with it, I would highly suggest taking a look at the [Integration Quality Scale](https://developers.home-assistant.io/docs/integration_quality_scale_index). It outlines different things you should take into consideration when creating or modifying an integration. Specifically on the Silver tier, it mentions "It will not print any exceptions nor will it fill the log with retry attempts." If possible, you should try to ensure as many of these specifications are satasfied.

Fair comment! I've amended the PR taking this in account.

OnFreund commented:
LGTM

emontnemery reviewed (approved):
Thanks, @cgtobi 👍
